frontend/javascript/modules/newYear.js

```diff
@@ -0,0 +1,63 @@
+import FetchUtil from '../utils/fetchUtil';
+
+export default class NewYear {
+  constructor(link) {
+    this.link = link;
+    this.step = this.link.dataset.step;
+    this.uid = this.link.dataset.uid;
+
+    this.endpoints = {
+      housing: '/housing',
+      active: '/manage/active',
+      current: '/manage/current/'
+    };
+
+    this.render();
+  }
+  render() {
+    this.link.addEventListener('click', e => {
+      e.preventDefault();
+
+      if (this.step === "welcome") {
+        $('#new-welcome').fadeOut(function() {
+          $("#new-clear").fadeIn();
+        });
+      } else if (this.step === "clear") {
+        FetchUtil.fetchWithWarning(this.endpoints.active, {
+          method: 'DELETE',
+          warningText: "This will clear active members and room assignments!",
+          successText: "Data successfully cleared."}, () => {
+            fetch(this.endpoints.housing, {
```

Comment:
You should have a `.catch()` attached to the promise to handle failure cases. Take a look at FetchUtil for an example.